assessment/models.py

```diff
@@ -712,12 +732,53 @@ def get_selected_option_content(self):
         return self.selected_option.get_content()
 
 
+class ResponseTestAttempt(ToolAttempt):
+    submitted_time = models.DateTimeField(default=None, null=True)
+    subject = models.TextField(null=True)
+    response = models.TextField(null=True)
+
+    def set_subject(self, subject):
+        self.subject = subject
+        self.save()
+
+    def set_response(self, response):
+        self.submitted_time = datetime.datetime.now(tz=pytz.utc)
+        self.response = response
+        self.save()
+
+
+class ResponseTestAttemptSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = ResponseTestAttempt
+        fields = ['tool_attempt_id', 'submitted_time', 'subject', 'response']
+
+
 class AssessmentEventParticipation(models.Model):
     assessment_event = models.ForeignKey('assessment.AssessmentEvent', on_delete=models.CASCADE)
     assessee = models.ForeignKey('users.Assessee', on_delete=models.CASCADE)
     assessor = models.ForeignKey(USERS_ASSESSOR, on_delete=models.RESTRICT)
     attempt = models.OneToOneField('assessment.TestFlowAttempt', on_delete=models.CASCADE, null=True)
 
+    def get_all_response_test_attempts(self):
+        return self.attempt.toolattempt_set.instance_of(ResponseTestAttempt)
+
+    def get_response_test_attempt(self, response_test: ResponseTest):
+        response_test_attempts = self.get_all_response_test_attempts()
+        matching_response_test_attempts = response_test_attempts.filter(assessment_tool_attempted=response_test)
+
+        if matching_response_test_attempts:
+            return matching_response_test_attempts[0]
+
+        else:
+            return None
+
+    def create_response_test_attempt(self, response_test: ResponseTest):
```

Comment:
Hello Jo, I would like to ask when this method will be called. Will it be called when I get the response test attempt for the first time or do I have to post to a separate endpoint to create the attempt? Let me know ;D

Reply:
Hello, Rashad. This will be called when the user first submit their response to a response test. I believe there is no need for us to create a separate endpoint to create the attempt.

Regards,
Johanes Christian

Reply:
Okay jo, Noted.